pkg/unstruct/v1alpha2/kubernetes.go

```diff
@@ -146,20 +180,150 @@ func (k *kubeclient) DeleteAll(u ...*unstructured.Unstructured) []error {
 
 // Create creates an unstructured instance at
 // kubernetes cluster
-func (k *kubeclient) Create(u *unstructured.Unstructured) error {
+func (k *Kubeclient) Create(u *unstructured.Unstructured, cOpts ...CreateOptionFn) error {
+	if u == nil {
+		return errors.Errorf("create failed: nil unstruct instance was provided")
+	}
 	cli, err := k.getClientOrCached()
 	if err != nil {
 		return err
 	}
-	return k.CreateFn(cli, context.TODO(), u)
+	opt := NewCreateOption(cOpts...)
+	_, err = k.create(cli, u, opt.createOptions, opt.subresources...)
+	return err
 }
 
 // Delete deletes the unstructured instance from
 // kubernetes cluster
-func (k *kubeclient) Delete(u *unstructured.Unstructured) error {
+func (k *Kubeclient) Delete(u *unstructured.Unstructured, dOpts ...DeleteOptionFn) error {
 	cli, err := k.getClientOrCached()
 	if err != nil {
 		return err
 	}
-	return k.DeleteFn(cli, context.TODO(), u)
+	opts := NewDeleteOption(dOpts...)
+	return k.delete(cli, u, opts.deleteOptions)
+}
+
+// GetOption holds the kubernetes options
+// to get a resource
+type GetOption struct {
+	namespace    string
+	getOption    metav1.GetOptions
+	grv          schema.GroupVersionResource
+	subresources []string
+}
+
+// GetOptionFn abstracts the construction of GetOption
+type GetOptionFn func(*GetOption)
```

Comment:
It will be better to avoid typed function here.
GetOptionBuilder struct will be a better choice.